src/Microsoft.ML.Data/Scorers/PredictionTransformer.cs

```diff
@@ -769,6 +769,13 @@ internal static object Create(IHostEnvironment env, ModelLoadContext ctx, IHost
 
             return genericInstance;
         }
+
+        internal static Type GetLoadType(Type modelType)
+        {
+            var y = modelType.GetField("LoadType", BindingFlags.NonPublic | BindingFlags.Static);
```

Comment:
This approach is very easily broken, since people won't know that the unused field you are adding is only being used through reflection.

Instead, I think you should just inspect the type using normal Reflection APIs. Check out these methods

* [GetGenericArguments()](https://docs.microsoft.com/en-us/dotnet/api/system.type.getgenericarguments?view=netframework-4.8)
* [GetGenericTypeDefinition()](https://docs.microsoft.com/en-us/dotnet/api/system.type.getgenerictypedefinition?view=netframework-4.8)

Reply:
Thanks for the feedback. But I am not sure how to follow your suggestion.

Since the problem is that when loading a `PMCMP<TSubmodel, TCalibrator>` inside a `BinaryPredictionTransformer<>`, what actually needs to be loaded is not a `BPT<PMCMP<>>` but rather a `BPT<CMPB<>>`, then are you suggesting something similar to the following pseudocode?:

```
// Inside the CreatePredictionTransformer class
internal static Type GetLoadType(Type modelType)
{
    if(modelType is PMCMP)
    {
         use GetGenericArguments() and GetGenericTypeDefinition() to get the correct
        parameter types for the submodel and calibrator, and return type CMPB with the
        the correct type parameters.
    }
     
     // if the type of the model isn't PMCMP then simply return the original type
     return modelType;
}
```

In short, is your suggestion to hardcode the case of PMCMP directly in the method that is called when loading a BPT, so that it gets managed differently to other cases??

Thanks.

Reply:
So I will resolve this conversation, because, as we discussed offline, it's better to use Attributes in this case, and I've already implemented that. Thanks.